Gradle v7 cannot gather test result

```bash
> Task :copySubProjectsTestResults FAILED
39 actionable tasks: 33 executed, 6 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':copySubProjectsTestResults'.
> Entry test/binary/output.bin is a duplicate but no duplicate handling strategy has been set. Please refer to 
https://docs.gradle.org/7.3.3/dsl/org.gradle.api.tasks.Copy.html#org.gradle.api.tasks.Copy:duplicatesStrategy for details.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -30,7 +30,6 @@ jobs:
       version: ${{ steps.context.outputs.version }}
       commitId: ${{ steps.context.outputs.shortCommitId }}
       semanticVersion: ${{ steps.dynamic_prop.outputs.semanticVersion }}
-      disablePlugin: ${{ steps.dynamic_prop.outputs.disablePlugin }}
 
     steps:
       - uses: actions/checkout@v2
@@ -49,7 +48,7 @@ jobs:
 
       - name: Project context
         id: context
-        uses: zero88/gh-project-context@v1.1.0
+        uses: zero88/gh-project-context@v1.1
         with:
           mustSign: true
           nextVerMode: PATCH

--- a/src/main/kotlin/io/zero88/gradle/OSSProjectPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/OSSProjectPlugin.kt
@@ -1,6 +1,7 @@
 package io.zero88.gradle
 
 import io.zero88.gradle.helper.computeBaseName
+import io.zero88.gradle.helper.checkMinGradleVersion
 import io.zero88.gradle.helper.prop
 import org.gradle.api.Plugin
 import org.gradle.api.Project
@@ -33,7 +34,14 @@ import java.util.jar.Attributes
 @Suppress("UnstableApiUsage")
 class OSSProjectPlugin : Plugin<Project> {
 
+    companion object {
+
+        const val PLUGIN_ID = "io.github.zero88.gradle.oss"
+    }
+
     override fun apply(project: Project) {
+        project.logger.info("Applying plugin '${PLUGIN_ID}'")
+        checkMinGradleVersion(PLUGIN_ID)
         applyExternalPlugins(project)
         val ossExt = evaluateProject(project)
         project.tasks {

--- a/src/main/kotlin/io/zero88/gradle/RootProjectPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/RootProjectPlugin.kt
@@ -1,5 +1,6 @@
 package io.zero88.gradle
 
+import io.zero88.gradle.helper.checkMinGradleVersion
 import io.zero88.gradle.helper.prop
 import org.gradle.api.Plugin
 import org.gradle.api.Project
@@ -32,12 +33,15 @@ class RootProjectPlugin : Plugin<Project> {
         const val COPY_SUB_PROJECT_TEST_RESULTS_TASK_NAME = "copySubProjectsTestResults"
         const val ROOT_TEST_REPORT_TASK_NAME = "testRootReport"
         const val ROOT_JACOCO_TASK_NAME = "jacocoRootReport"
+        const val PLUGIN_ID: String = "io.github.zero88.gradle.root"
     }
 
     override fun apply(project: Project) {
         if (project != project.rootProject) {
             return
         }
+        project.logger.info("Applying plugin '${PLUGIN_ID}'")
+        checkMinGradleVersion(PLUGIN_ID)
         applyExternalPlugins(project)
         configureExtension(project)
         if (isSingle(project)) {

--- a/src/main/kotlin/io/zero88/gradle/RootProjectPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/RootProjectPlugin.kt
@@ -3,6 +3,7 @@ package io.zero88.gradle
 import io.zero88.gradle.helper.prop
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.file.DuplicatesStrategy
 import org.gradle.api.plugins.JavaPluginConvention
 import org.gradle.api.publish.tasks.GenerateModuleMetadata
 import org.gradle.api.tasks.Copy
@@ -93,6 +94,7 @@ class RootProjectPlugin : Plugin<Project> {
         register<Copy>(COPY_SUB_PROJECT_ARTIFACTS_TASK_NAME) {
             group = "distribution"
             description = "Gathers sub projects artifacts"
+            duplicatesStrategy = DuplicatesStrategy.FAIL
             onlyIf { !isSingle(project) }
             dependsOn(project.subprojects.mapNotNull { it.tasks.findByName(LifecycleBasePlugin.ASSEMBLE_TASK_NAME) })
             from(project.subprojects.fold(listOf<File>()) { r, p -> r.plus(p.buildDir.resolve("distributions")) })
@@ -108,6 +110,7 @@ class RootProjectPlugin : Plugin<Project> {
         register<Copy>(COPY_SUB_PROJECT_TEST_RESULTS_TASK_NAME) {
             group = LifecycleBasePlugin.VERIFICATION_GROUP
             description = "Gathers sub projects test result"
+            duplicatesStrategy = DuplicatesStrategy.WARN
             dependsOn(project.subprojects.mapNotNull { it.tasks.withType<Test>() })
             from(project.subprojects.fold(listOf<File>()) { r, p -> r.plus(p.buildDir.resolve(TestingBasePlugin.TEST_RESULTS_DIR_NAME)) })
             into(project.buildDir.resolve(TestingBasePlugin.TEST_RESULTS_DIR_NAME))

--- a/src/main/kotlin/io/zero88/gradle/helper/GradleVersionChecker.kt
+++ b/src/main/kotlin/io/zero88/gradle/helper/GradleVersionChecker.kt
@@ -1,0 +1,14 @@
+package io.zero88.gradle.helper
+
+import org.gradle.tooling.UnsupportedVersionException
+import org.gradle.util.GradleVersion
+
+private val MIN_GRADLE_VERSION = GradleVersion.version("6.5")
+
+@Throws(UnsupportedVersionException::class)
+fun checkMinGradleVersion(pluginId: String) {
+    if (GradleVersion.current() >= MIN_GRADLE_VERSION) {
+        return
+    }
+    throw UnsupportedVersionException("The plugin '${pluginId}' requires at least Gradle ${MIN_GRADLE_VERSION.version} to be run.")
+}

--- a/src/main/kotlin/io/zero88/gradle/qwe/QWEDecoratorPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/qwe/QWEDecoratorPlugin.kt
@@ -1,7 +1,7 @@
 package io.zero88.gradle.qwe
 
 import io.zero88.gradle.OSSExtension
-import io.zero88.gradle.helper.prop
+import io.zero88.gradle.helper.checkMinGradleVersion
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.create
@@ -12,11 +12,15 @@ import org.gradle.kotlin.dsl.getByType
 interface QWEDecoratorPlugin<T> : Plugin<Project> {
 
     override fun apply(project: Project) {
+        project.logger.info("Applying plugin '${pluginId()}'")
+        checkMinGradleVersion(pluginId())
         applyExternalPlugins(project)
         val ossExt = project.extensions.getByType<OSSExtension>()
         val qweExt = project.extensions.findByType<QWEExtension>() ?: project.extensions.create(QWEExtension.NAME)
         registerAndConfigureTask(project, ossExt, qweExt, configureExtension(project, ossExt, qweExt))
     }
+
+    fun pluginId(): String
 
     fun applyExternalPlugins(project: Project)
 

--- a/src/main/kotlin/io/zero88/gradle/qwe/app/QWEAppPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/qwe/app/QWEAppPlugin.kt
@@ -35,10 +35,15 @@ class QWEAppPlugin : QWEDecoratorPlugin<QWEAppExtension> {
         const val FAT_JAR_CLASSIFIER = "fat"
         const val DIST_ZIP_FAT_TASK_NAME = "distZipFat"
         const val DIST_TAR_FAT_TASK_NAME = "distTarFat"
+        const val PLUGIN_ID = "io.github.zero88.gradle.qwe.app"
     }
 
     override fun applyExternalPlugins(project: Project) {
         project.plugins.apply(OSSProjectPlugin::class.java)
+    }
+
+    override fun pluginId(): String {
+        return PLUGIN_ID
     }
 
     override fun configureExtension(project: Project, ossExt: OSSExtension, qweExt: QWEExtension): QWEAppExtension {

--- a/src/main/kotlin/io/zero88/gradle/qwe/docker/QWEDockerPlugin.kt
+++ b/src/main/kotlin/io/zero88/gradle/qwe/docker/QWEDockerPlugin.kt
@@ -26,11 +26,16 @@ import org.gradle.language.base.plugins.LifecycleBasePlugin
         const val ARG_DOCKER_REGISTRY = "dockerRegistries"
         const val ARG_DOCKER_TAGS = "dockerTags"
         const val ARG_DOCKER_LABELS = "dockerLabels"
+        const val PLUGIN_ID = "io.github.zero88.gradle.qwe.docker"
     }
 
     override fun applyExternalPlugins(project: Project) {
         project.plugins.apply(DockerRemoteApiPlugin::class.java)
         project.plugins.apply(QWEAppPlugin::class.java)
+    }
+
+    override fun pluginId(): String {
+        return PLUGIN_ID
     }
 
     override fun configureExtension(project: Project, ossExt: OSSExtension, qweExt: QWEExtension): QWEDockerExtension {

